### PR TITLE
Ignore benign torch inductor TF32 advisory warning in CI

### LINKED ISSUE #6849: CI emits torch inductor UserWarning: TensorFloat32 tensor cores for float32 matrix multiplication available but not enabled
### What

CI emits a new `UserWarning` from `torch._inductor` in the Liger tests:

```
tests/test_dpo_trainer.py::TestDPOTrainerVLM::test_train_vlm_liger
tests/test_kto_trainer.py::TestKTOTrainer::test_train_with_liger
tests/test_dpo_trainer.py::TestDPOTrainer::test_train_with_liger
tests/test_kto_trainer.py::TestKTOTrainerVLM::test_train_vlm_liger
  /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/_inductor/compile_fx.py:321: UserWarning: TensorFloat32 tensor cores for float32 matrix multiplication available but not enabled. Consider setting `torch.set_float32_matmul_precision('high')` for better performance.
    warnings.warn(
```

CI run: https://github.com/huggingface/trl/actions/runs/32414561958/job/96572632322

Affected tests:

- `tests/test_dpo_trainer.py::TestDPOTrainer::test_train_with_liger`
- `tests/test_dpo_trainer.py::TestDPOTrainerVLM::test_train_vlm_liger`
- `tests/test_kto_trainer.py::TestKTOTrainer::test_train_with_liger`
- `tests/test_kto_trainer.py::TestKTOTrainerVLM::test_train_vlm_liger`

The tests themselves pass; for now this is warning noise only.

### When it started

First seen in the run for #6741 (`1701d3e`), which moved the GPU CI from T4 to L40S: https://github.com/huggingface/trl/actions/runs/32366392208/job/96416712686 (4 warnings, all of them this one). No warning was reported in the run for `06c89977`, on the previous runner: https://github.com/huggingface/trl/actions/runs/32348545011/job/96362435182.

### Next steps

Investigate the root cause, then decide whether anything is actionable on the TRL side and/or should be reported upstream.

### LINKED ISSUE #175484: Unable to disable warning about tf32 precision
### 🐛 Describe the bug

This message is repeated once per GPU in all my runs:
```
/usr/local/lib/python3.10/dist-packages/torch/_inductor/compile_fx.py:321: UserWarning: TensorFloat32 tensor cores for float32 matrix multiplication available but not enabled. Consider setting `torch.set_float32_matmul_precision('high')` for better performance.
  warnings.warn(
```
I want my fp32 matrix multiplications to be in fp32, not tf32. But there is no avenue to suppressing this warning besides filtering logs.

`torch.set_float32_matmul_precision("highest")` does not provide a way to distinguish the default "highest" from a user-set "highest". So maybe we could add a flag `matmul_precision_set_manually` that `_warn_tf32_disabled()` would check.

### Versions

```
PyTorch version: 2.10.0
Is debug build: False
CUDA used to build PyTorch: 12.9
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04.2) 11.4.0
Clang version: Could not collect
CMake version: version 4.2.1
Libc version: glibc-2.35

Python version: 3.10.12 (main, Jan  8 2026, 06:52:19) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-6.5.13-65-650-4141-22041-coreweave-amd64-85c45edc-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.9.86
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: NVIDIA H100 80GB HBM3
Nvidia driver version: 570.172.08
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.9.15.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv.so.9.15.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn.so.9.15.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_precompiled.so.9.15.1
/usr/lib/x86_64-linux-gnu/libcudnn_engines_runtime_compiled.so.9.15.1
/usr/lib/x86_64-linux-gnu/libcudnn_graph.so.9.15.1
/usr/lib/x86_64-linux-gnu/libcudnn_heuristic.so.9.15.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops.so.9.15.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      46 bits physical, 57 bits virtual
Byte Order:                         Little Endian
CPU(s):                             128
On-line CPU(s) list:                0-127
Vendor ID:                          GenuineIntel
Model name:                         Intel(R) Xeon(R) Platinum 8462Y+
CPU family:                         6
Model:                              143
Thread(s) per core:                 2
Core(s) per socket:                 32
Socket(s):                          2
Stepping:                           8
CPU max MHz:                        4100.0000
CPU min MHz:                        800.0000
BogoMIPS:                           5600.00
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hfi vnmi avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr ibt amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
Virtualization:                     VT-x
L1d cache:                          3 MiB (64 instances)
L1i cache:                          2 MiB (64 instances)
L2 cache:                           128 MiB (64 instances)
L3 cache:                           120 MiB (2 instances)
NUMA node(s):                       2
NUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94,96,98,100,102,104,106,108,110,112,114,116,118,120,122,124,126
NUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95,97,99,101,103,105,107,109,111,113,115,117,119,121,123,125,127
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] clip-anytorch==2.6.0
[pip3] dctorch==0.1.2
[pip3] DISTS_pytorch==0.1
[pip3] gpytorch==1.15.1
[pip3] lovely-numpy==0.2.18
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.24.4
[pip3] torch==2.10.0
[pip3] torchaudio==2.10.0
[pip3] torchdata==0.11.0
[pip3] torchdiffeq==0.2.5
[pip3] torchsde==0.2.6
[pip3] torchtitan==0.1.0
[pip3] torchvision==0.25.0
[pip3] triton==3.6.0+git8fedd49b
[pip3] welford-torch==0.2.5
[conda] Could not collect
```

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by albertvillanova**
This PR silences the benign `torch._inductor` TF32 advisory `UserWarning` emitted by the Liger tests in CI.

Fix #6849.

### Motivation

Since #6741 moved the GPU CI from T4 to L40S, the DPO and KTO Liger tests emit:

```
tests/test_dpo_trainer.py::TestDPOTrainerVLM::test_train_vlm_liger
tests/test_kto_trainer.py::TestKTOTrainer::test_train_with_liger
tests/test_dpo_trainer.py::TestDPOTrainer::test_train_with_liger
tests/test_kto_trainer.py::TestKTOTrainerVLM::test_train_vlm_liger
  /__w/trl/trl/.venv/lib/python3.12/site-packages/torch/_inductor/compile_fx.py:321: UserWarning: TensorFloat32 tensor cores for float32 matrix multiplication available but not enabled. Consider setting `torch.set_float32_matmul_precision('high')` for better performance.
    warnings.warn(
```

Inductor emits this advisory on every compile when TF32 tensor cores exist and the float32 matmul precision is left at its default. It is guarded on compute capability `>= (8, 0)`, so it was unreachable on T4 (sm75) and now fires on L40S (sm89). The four Liger tests are the only ones that reach an inductor compile, through the `torch.compile` in Liger's chunked-loss base classes.

Nothing is actionable on the TRL side: what the advisory asks for, `torch.set_float32_matmul_precision('high')`, is a process-wide numerics setting that a library must not change for its users, and setting it in the test session would alter the numerics of every float32 matmul under our tolerance-based training tests. The message is informational only, compilation and execution succeed either way, and torch offers no way to opt out, since it cannot distinguish the default precision from a user-set one.

Already reported upstream by others, neither fixed: https://github.com/pytorch/pytorch/issues/175484 asks for a way to opt out, and https://github.com/pytorch/pytorch/pull/185541 proposed demoting the advisory to `log.info()`.

### Solution

Suppress the warning through `filterwarnings` in `pyproject.toml`, documented like the other upstream warning filters and linked to the tracking and upstream issues, so the entry can be removed once no supported `torch` version emits it.

### Changes

- Add a `filterwarnings` entry for the inductor TF32 advisory `UserWarning`, with a comment explaining the cause, what triggers it, and links to the tracking and upstream issues

---

> [!NOTE]
> **Low Risk**
> CI-only pytest warning filter; no runtime, training, or security behavior changes.
> 
> **Overview**
> Suppresses a benign torch inductor `UserWarning` about TF32 tensor cores not being enabled, which started firing in GPU CI after the T4 → L40S move.
> 
> Adds a documented `filterwarnings` entry in `pyproject.toml` so DPO/KTO Liger tests (the only ones that `torch.compile`) stay clean without changing process-wide matmul precision.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f80bee61499aa918e87ea81b0fe33ef2bc56d75f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by albertvillanova** [comment]
Note: #6850 appends to the end of the same `filterwarnings` list, so whichever of the two merges second will need a trivial rebase.

**Comment by albertvillanova** [comment]
Merging as hotfix.
